### PR TITLE
PUD-712: Automatically update return to custody status

### PR DESCRIPTION
**Pull request by philsegal**
- Upon request to get all recalls, query the prisoner search endpoint for any recalls 'AWAITING_RETURN_TO_CUSTODY'
 - For any offenders returned to custody, update their RTC record and calculate dossierTargetDate.
 - returnToCustody.updateThresholdMinutes property to allow E2E tests to have a window to manually update RTC as we are unable to control the status of offenders in dev/preprod
 - included in health endpoint

### CODE REVIEW

File: src/test/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/service/RecallServiceTest.kt [hunk 1]
```diff
@@ -308,6 +340,88 @@ class RecallServiceTest {
...
+    every { recallRepository.findAll() } returns recallList
+    every { prisonerOffenderSearchClient.prisonerByNomsNumber(nicStillUalPreviouslyUpdatedNoms) } returns Mono.just(nicStillUalRecentlyUpdatedPrisoner)
+    every { prisonerOffenderSearchClient.prisonerByNomsNumber(rtcNoms) } returns Mono.just(nicRtcPrisoner)
+
+    every { prisonApiClient.latestInboundMovements(setOf(rtcNoms)) } returns listOf(Movement(rtcNoms.value, movementDate, movementTime))
+    every { recallRepository.save(expectedRecall, RecallService.SYSTEM_USER_ID) } returns expectedRecall
+
+    underTest.updateCustodyStatus(currentUserId)
+
+    verify { recallRepository.save(expectedRecall, RecallService.SYSTEM_USER_ID) }
```
Comment (AndyNewton) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/service/RecallServiceTest.kt:422]: Wondering if any further `verify` calls might be justified?  E.g. to `bankHolidayService`?  
🤔  Would this test fail if zero calls were made matching the `every { bankHolidayService.nextWorkingDate ...` etc. calls?

**Review by AndyNewton — DISMISSED** [review]
Basically, 1 @Transactional missing?, plus a few other QQs.

File: src/main/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/health/PingHealthCheck.kt [hunk 2]
```diff
@@ -17,7 +17,7 @@ abstract class PingHealthCheck(
   private val webClient: WebClient,
   private val componentName: String,
   private val healthUrl: String,
-  private val timeout: Duration = Duration.ofSeconds(2)
+  private val timeout: Duration = Duration.ofSeconds(1)
```
[the remark this answers is not in the document]
Comment (AndyNewton) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/health/PingHealthCheck.kt:20]: OK; makes the tests run n x 1 seconds faster ...

[the remark this answers is not in the document]
Comment (philsegal) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/health/PingHealthCheck.kt:20]: Yes, that and that there is a 15s limit and we were hitting it by having so many timeouts being tested.

File: src/test/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/service/RecallServiceTest.kt (hunk 1 above)
Comment (philsegal) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/service/RecallServiceTest.kt:422]: It's annoying you cant force the every to also verify

**Review by AndyNewton — DISMISSED** [review]

**Review by AndyNewton — APPROVED** [review]

**Pull request merged**